### PR TITLE
Ln cted 2289

### PR DESCRIPTION
**Pull request by loyisongq**

### CODE REVIEW

File: classes/observer.php [hunk 1]
```diff
@@ -172,11 +172,12 @@ public static function email_mention_hsu(\mod_hsuforum\event\assessable_uploaded
                         $discussion_id = $other->discussionid;
                         $post_id = $event->objectid;
                         $course_id = $event->courseid;
+                        $reply_id = $event->objectid;
```
Comment (BrendonPretorius) [inline classes/observer.php:175]: Is this correct? See it is also being used for post_id

File: classes/observer.php [hunk 2]
```diff
@@ -172,11 +172,12 @@ public static function email_mention_hsu(\mod_hsuforum\event\assessable_uploaded
...
 
                         $course_name = $DB->get_field("course", "fullname", array("id"=>$course_id));
                         $from_user = $DB->get_record("user", array("id"=>$event->userid));
 
-                        $link = $_SERVER['HTTP_HOST'] . '/mod/hsuforum/discuss.php?d=' . $discussion_id . '&postid' . $post_id . '#p' . $post_id;
+                        $link = $_SERVER['HTTP_HOST'] . '/blocks/hsuforum_users/forumusers.php?courseid=' . $course_id . '&discussion=' . $discussion_id . '&reply=' . $reply_id;
```
Comment (BrendonPretorius) [inline classes/observer.php:180]: Any reason for not using moodleurl?

File: classes/observer.php [hunk 3]
```diff
@@ -203,7 +204,7 @@ public static function email_mention_hsu(\mod_hsuforum\event\assessable_uploaded
 
                         $eventdata->customdata = $customdata;
 
-                        $contexturl = new moodle_url('/mod/hsuforum/discuss.php', array('d' => $discussion_id, 'postid' => $post_id), 'p' . $post_id);
+                        $contexturl = new moodle_url('/blocks/hsuforum_users/forumusers.php', array('courseid' => $course_id, 'discussion' => $discussion_id, 'reply' => $reply_id));
```
Comment (BrendonPretorius) [inline classes/observer.php:207]: Same as above

**Review by BrendonPretorius — CHANGES_REQUESTED** [review]

File: classes/observer.php (hunk 1 above)
Comment (loyisongq) [reply classes/observer.php:175]: shot changed this

**Review by BrendonPretorius — APPROVED** [review]

**Pull request merged**